security/pkg/server/ca/server.go

```diff
@@ -78,12 +78,14 @@ func (s *Server) CreateCertificate(ctx context.Context, request *pb.IstioCertifi
 	}
 
 	// TODO: Call authorizer.
-
+	crMetadata := request.Metadata.GetFields()
+	certSigner := crMetadata[security.CertSigner].GetStringValue()
```

Comment:
Is there a chance that this could be an empty string? Is that case handled?

Reply:
Yes, we handled this case:  will use the value from `r.raOpts.CaSigner` if it is an empty string like below
```
	if certSigner == "" {
		certSigner = r.raOpts.CertSignerDomain + "/" + r.raOpts.CaSigner
	} else {
		certSigner = r.raOpts.CertSignerDomain + "/" + certSigner
	}
```